src/plugins/data/public/search/search_service.ts

```diff
@@ -271,6 +270,9 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
       showError: (e: Error) => {
         this.searchInterceptor.showError(e);
       },
+      showWarnings: (warning: ResponseWarning) => {
+        this.searchInterceptor.showWarning(warning);
```

Comment:
seems you could implement this here directly instead of using search interceptor ?

Reply:
We need to add a class member for the `notifications` dependency in order to show a toast directly here. On the other hand, the `addWarning` method is similar enough conceptually with `addError`, and the search intercepter has a class member for notifications to be shown from a method.

WDYT?

Reply:
Sorry @ppisljar --- I had no idea how to implement this method, but in the current code I am using the optional consumer callback, and/or calling `handleResponse` from here